src/server/pps/cmds/cmds.go

```diff
@@ -393,26 +394,47 @@ each datum.`,
 			if page < 0 {
 				return errors.Errorf("page must be zero or positive")
 			}
-			if raw {
+			var printF func(*ppsclient.DatumInfo) error
+			if !raw {
+				if output != "" {
+					cmdutil.ErrorAndExit("cannot set --output (-o) without --raw")
+				}
+				writer := tabwriter.NewWriter(os.Stdout, pretty.DatumHeader)
+				printF = func(di *ppsclient.DatumInfo) error {
+					pretty.PrintDatumInfo(writer, di)
+					return nil
+				}
+				defer func() {
+					if err := writer.Flush(); retErr == nil {
+						retErr = err
+					}
+				}()
+			} else {
 				e := encoder(output)
-				return client.ListDatumF(args[0], pageSize, page, func(di *ppsclient.DatumInfo) error {
+				printF = func(di *ppsclient.DatumInfo) error {
 					return e.EncodeProto(di)
-				})
-			} else if output != "" {
-				cmdutil.ErrorAndExit("cannot set --output (-o) without --raw")
+				}
 			}
-			writer := tabwriter.NewWriter(os.Stdout, pretty.DatumHeader)
-			if err := client.ListDatumF(args[0], pageSize, page, func(di *ppsclient.DatumInfo) error {
-				pretty.PrintDatumInfo(writer, di)
-				return nil
-			}); err != nil {
-				return err
+			if pipelineInputPath != "" {
+				pipelineReader, err := ppsutil.NewPipelineManifestReader(pipelineInputPath)
+				if err != nil {
+					return err
+				}
+				request, err := pipelineReader.NextCreatePipelineRequest()
+				if err != nil {
+					return err
+				}
+				return client.ListDatumInputF(request.Input, pageSize, page, printF)
+			} else if len(args) == 1 {
+				return client.ListDatumF(args[0], pageSize, page, printF)
+			} else {
+				return errors.Errorf("must specify either a job or a pipeline spec")
 			}
-			return writer.Flush()
 		}),
 	}
 	listDatum.Flags().Int64Var(&pageSize, "pageSize", 0, "Specify the number of results sent back in a single page")
 	listDatum.Flags().Int64Var(&page, "page", 0, "Specify the page of results to send")
+	listDatum.Flags().StringVarP(&pipelineInputPath, "file", "f", "", "The JSON file containing the pipeline to list datums from, the pipeline need not exist.")
```

Comment:
nit: the other flags don't use periods and this isn't a full sentence - drop the period